include/llama/Copy.hpp

```diff
@@ -0,0 +1,364 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include "View.hpp"
+#include "mapping/AoSoA.hpp"
+#include "mapping/SoA.hpp"
+
+#include <cstring>
+#include <numeric>
+
+namespace llama
+{
+    namespace internal
+    {
+        inline void parallel_memcpy(
+            std::byte* dst,
+            const std::byte* src,
+            std::size_t size,
+            std::size_t threadId = 0,
+            std::size_t threadCount = 1,
+            decltype(std::memcpy) memcpy = std::memcpy)
```

Comment:
Is the `decltype(std::memcpy)` laziness or does it have a deeper meaning?

Reply:
`decltype(std::memcpy)` is a shorter way to spell the function pointer type for `memcpy` :) This parameter is intended to provide a custom implementation for `memcpy`. E.g. the AVX2 `memcpy` in the viewcopy example.

Reply:
Renamed.